### ISSUE TITLE
I also did some investigations...

### ISSUE THREAD

**Issue by darkman1983** [guedou/TS-453Be#1]
Hi there,
i currently try getting the temps on my System.
I have a QNAP TS-1677x and did some digging on Google about this whole thing.
Just found your Repo here, but can't seem to contact you directly.

Seems my NAS has the same chip than yours, im also on linux, but on unRaid rather than Ubuntu.
I only get this temps:

`root@DarkSpace:~# sensors
k10temp-pci-00c3
Adapter: PCI adapter
CPU Temp:     +40.0°C  (high = +70.0°C)
Tctl:         +40.0°C
`

So i tried sensors-detect, which bought up this:
`Some Super I/O chips contain embedded sensors. We have to write to
standard I/O ports to probe them. This is usually safe.
Do you want to scan for Super I/O sensors? (YES/no):
Probing for Super-I/O at 0x2e/0x2f
Trying family National Semiconductor/ITE'...               Yes
Found unknown chip with ID 0x8528
`
Did you get your temps in first instance or did you get the same?
After that i did some more digging on Google and came up with some links and information, because the 0x8528 seems to be the it8528, so here is all i found:
- There should be a kernel patch for that driver, because i found this... https://lore.kernel.org/patchwork/patch/634176/ or it never made it into the kernel itself...
- As i found this, i did some more digging, and also found this... https://lwn.net/Articles/705319/
- And also this... https://github.com/rvido/iManager

I searched my kernel modules, but it wasn't available, than i tested availability in Kernel 5.x, but it seems this is noch standard module and must be enabled on kernel compile.

So maybe we need only the compiled driver loaded and could get all those things working. Maybe we need no more reversing...?

I hope that this is somewhat usefull.

**Comment by guedou** [comment]
Thanks for your message. The lack of support for IT8528 is the reason why I wrote the pnaq tool. Did you try it?

**Comment by darkman1983** [comment]
Currently not, because i'm totally new to unRaid and have no experience with Linux on base of Slackware. I have to rebuild the kernel with this enabled, so that the modules are build, they won't be a default.

**Comment by guedou** [comment]
Did you manage to make this work?

**Comment by scottshanafelt** [comment]
The panq tool works fine on unraid, no changes to kernel or kernel params necessary.  But a note to others you are best to build it on another system (I used Arch).  Getting GCC and the other required dev tools setup on unraid is not necessary (if it is even possible, I dunno).

**Comment by guedou** [comment]
@scottshanafelt thanks for letting us know! Did you end up compiling a static binary and copying it to your unraid NAS?

**Comment by scottshanafelt** [comment]
yep exactly.  I did end up forking to make modifications for my qnap ts-877, but your core methods supported multiple IDs already, I just exposed more commands.  Thanks!
